lib/spack/spack/binary_distribution.py

```diff
@@ -1401,32 +1401,44 @@ def nodes_to_be_packaged(specs, root=True, dependencies=True):
         return list(filter(packageable, nodes))
 
 
-def push(specs, push_url, include_root: bool = True, include_dependencies: bool = True, **kwargs):
-    """Create a binary package for each of the specs passed as input and push them
-    to a given push URL.
+def push(
+    spec: Spec,
+    mirror_url,
+    force=False,
+    relative=False,
+    unsigned=False,
+    allow_root=False,
+    key=None,
+    regenerate_index=False,
+):
+    """Create and push binary package for a single spec to the specified
+    mirror url.
 
     Args:
-        specs (List[spack.spec.Spec]): installed specs to be packaged
-        push_url (str): url where to push the binary package
-        include_root (bool): include the root of each spec in the nodes
-        include_dependencies (bool): include the dependencies of each
-            spec in the nodes
-        **kwargs: TODO
+        spec (spack.spec.Spec): Spec to package and push
+        mirror_url (str): Desired destination url for binary package
+        kwargs: Passed through to safe_push(...)
```

Comment:
There are no `**kwargs` so not sure how to write the docstring in a concise form... I think it would be worthwhile to gather these many arguments into a coherent structure (a sort of data class?), bu maybe this is for another PR.

Reply:
Also here we can add type-hints to signature and remove them from the docstring